x/examples/test-connectivity/main.go

```diff
@@ -282,10 +314,46 @@ func main() {
 					tcpReports = append(tcpReports, report)
 					mu.Unlock()
 				}
-				return newTCPTraceDialer(onDNS, onDial).DialStream(ctx, addr)
+				onDialStart := func(ctx context.Context, network, addr string) {
```

Comment:
This should save the start time, and return the onDone handler, which can refer to the start time directly.